core/src/io/anuke/mindustry/ui/dialogs/CustomRulesDialog.java

```diff
@@ -1,67 +1,106 @@
 import io.anuke.mindustry.core.Platform;
+import io.anuke.mindustry.game.Gamemode;
 import io.anuke.mindustry.game.Rules;
+import io.anuke.mindustry.graphics.Pal;
 
 import static io.anuke.mindustry.Vars.tilesize;
+import static io.anuke.mindustry.Vars.ui;
 
 public class CustomRulesDialog extends FloatingDialog{
-    public final Rules rules = new Rules();
     private Table main;
+    public Rules rules;
+    public Gamemode selectedGamemode;
 
     public CustomRulesDialog(){
         super("$mode.custom");
 
-        rules.waves = true;
-        rules.waveTimer = true;
-
         setFillParent(true);
         shown(this::setup);
         addCloseButton();
     }
 
+    public void show(Rules rules, Gamemode gamemode){this.rules = rules; this.selectedGamemode = gamemode; show();}
+
     void setup(){
         cont.clear();
         cont.pane(m -> main = m);
         main.margin(10f);
-
+        main.addButton("$rules.restoreDefault", () -> {rules = selectedGamemode.get(); setup();}).size(300f, 50f);
         main.left().defaults().fillX().left().pad(5);
         main.row();
-        main.addCheck("$rules.infiniteresources", b -> rules.infiniteResources = b).checked(b -> rules.infiniteResources);
-        main.row();
-        main.addCheck("$rules.wavetimer", b -> rules.waveTimer = b).checked(b -> rules.waveTimer);
-        main.row();
-        main.addCheck("$rules.waves", b -> rules.waves = b).checked(b -> rules.waves);
-        main.row();
-        main.addCheck("$rules.pvp", b -> rules.pvp = b).checked(b -> rules.pvp);
-        main.row();
-        main.addCheck("$rules.unitdrops", b -> rules.unitDrops = b).checked(b -> rules.unitDrops);
-        main.row();
-        number("$rules.buildcostmultiplier", f -> rules.buildCostMultiplier = f, () -> rules.buildCostMultiplier);
-        number("$rules.buildspeedmultiplier", f -> rules.buildSpeedMultiplier = f, () -> rules.buildSpeedMultiplier);
-        number("$rules.unitbuildspeedmultiplier", f -> rules.unitBuildSpeedMultiplier = f, () -> rules.unitBuildSpeedMultiplier);
-        number("$rules.unithealthmultiplier", f -> rules.unitHealthMultiplier = f, () -> rules.unitHealthMultiplier);
-        number("$rules.playerhealthmultiplier", f -> rules.playerHealthMultiplier = f, () -> rules.playerHealthMultiplier);
-        number("$rules.playerdamagemultiplier", f -> rules.playerDamageMultiplier = f, () -> rules.playerDamageMultiplier);
-        number("$rules.unitdamagemultiplier", f -> rules.unitDamageMultiplier = f, () -> rules.unitDamageMultiplier);
-        number("$rules.enemycorebuildradius", f -> rules.enemyCoreBuildRadius = f * tilesize, () -> Math.min(rules.enemyCoreBuildRadius / tilesize, 200));
-        number("$rules.respawntime", f -> rules.respawnTime = f * 60f, () -> rules.respawnTime / 60f);
-        number("$rules.wavespacing", f -> rules.waveSpacing = f * 60f, () -> rules.waveSpacing / 60f);
+        title("$rules.title.waves", Gamemode.survival);
```

Comment:
There's no way to configure `manyCores` here. This makes it impossible to play an attack-based custom gamemode.

Reply:
But wait. It's set by default when you select Attack gamemode. New custom-rules screen only allows you to modify rules which don't effect objective. That's why RulePreset is now Gamemode (they have different goals)